tests/release/e2e-test-push-image-to-pyxis.go

```diff
@@ -147,68 +159,128 @@ var _ = framework.ReleaseSuiteDescribe("[HACBS-1571]test-release-e2e-push-image-
-				if prList == nil || err != nil || len(prList.Items) < 1 {
+
+				releasePr, err := fw.AsKubeAdmin.TektonController.GetPipelineRun(releasePrName, managedNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("\nError getting Release PirpelineRun %s:\n %s", releasePr, err)
+					return false
+				}
+				additionalRleasePr, err := fw.AsKubeAdmin.TektonController.GetPipelineRun(additionalReleasePrName, managedNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("\nError getting PirpelineRun %s:\n %s", additionalRleasePr, err)
 					return false
 				}
 
-				return prList.Items[0].HasStarted() && prList.Items[0].IsDone() && prList.Items[0].Status.GetCondition(apis.ConditionSucceeded).IsTrue()
+				return releasePr.HasStarted() && releasePr.IsDone() && releasePr.Status.GetCondition(apis.ConditionSucceeded).IsTrue() &&
+					additionalRleasePr.HasStarted() && additionalRleasePr.IsDone() && additionalRleasePr.Status.GetCondition(apis.ConditionSucceeded).IsTrue()
 			}, releasePipelineRunCompletionTimeout, defaultInterval).Should(BeTrue())
 		})
 
-		It("validate the result of task create-pyxis-image contains id and succeeded.", func() {
+		It("validate the result of task create-pyxis-image contains image ids.", func() {
 			Eventually(func() bool {
-				prList, err := fw.AsKubeAdmin.TektonController.ListAllPipelineRuns(managedNamespace)
-				if prList == nil || err != nil || len(prList.Items) < 1 {
+
+				releasePr, err := fw.AsKubeAdmin.TektonController.GetPipelineRun(releasePrName, managedNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("\nError getting Release PirpelineRun %s:\n %s", releasePr, err)
+					return false
+				}
+				additionalRleasePr, err := fw.AsKubeAdmin.TektonController.GetPipelineRun(additionalReleasePrName, managedNamespace)
+				if err != nil {
+					GinkgoWriter.Printf("\nError getting PirpelineRun %s:\n %s", additionalRleasePr, err)
 					return false
 				}
+				re := regexp.MustCompile("[a-fA-F0-9]{24}")
 
-				pr, err := kubeController.Tektonctrl.GetPipelineRun(prList.Items[0].Name, managedNamespace)
-				Expect(err).NotTo(HaveOccurred())
+				trReleasePr, err := kubeController.GetTaskRunStatus(fw.AsKubeAdmin.CommonController.KubeRest(), releasePr, "create-pyxis-image")
+				if err != nil {
+					Expect(err).NotTo(HaveOccurred())
+				}
 
-				tr, err := kubeController.GetTaskRunStatus(fw.AsKubeAdmin.CommonController.KubeRest(), pr, "create-pyxis-image")
-				Expect(err).NotTo(HaveOccurred())
+				trAdditionalReleasePr, err := kubeController.GetTaskRunStatus(fw.AsKubeAdmin.CommonController.KubeRest(), additionalRleasePr, "create-pyxis-image")
+				if err != nil {
+					Expect(err).NotTo(HaveOccurred())
+				}
 
-				re := regexp.MustCompile("^[0-9a-z]{10,50}")
-				returnedImageID := re.FindString(tr.Status.TaskRunResults[0].Value.StringVal)
+				if len(re.FindAllString(trReleasePr.Status.TaskRunResults[0].Value.StringVal, -1)) >
```

Comment:
imo before this condition there should be a check for the length of `trReleasePr.Status.TaskRunResults`, otherwise it can end up in runtime error here

Reply:
Agree for the condition of length in taskRun results, 

for the second part, yes using `strings.split`  would work but what is the benefit instead  using the regex as current ?

Reply:
keeps the code easier to read - I would use regex only when it's absolutely necessary, which is not in this case IMO.

Reply:
if you're sticking with the `re.FindAllString()`, I think it's better to use it just once, store the result in a variable and then use the variable in following conditions instead (as suggested in the other comment)